listenbrainz/spotify_updater/spotify_read_listens.py

```diff
@@ -267,12 +264,23 @@ def process_one_user(user):
     if user.token_expired:
         try:
             user = spotify.refresh_user_token(user)
-        except spotify.SpotifyAPIError:
-            current_app.logger.error('Could not refresh user token from spotify', exc_info=True)
-            raise
+        except spotify.SpotifyAPIError as e:
+            # if it is an error from the Spotify API, show the error message to the user
+            spotify.update_last_updated(
+                user_id=user.user_id,
+                success=False,
+                error_message=str(e),
+            )
+            if not current_app.config['TESTING']:
+                notify_error(user.musicbrainz_row_id, str(e))
+            raise spotify.SpotifyListenBrainzError("Could not refresh user token from spotify")
         except spotify.SpotifyInvalidGrantError:
-            current_app.logger.error("%s has revoked spotify authorization", str(user), exc_info=True)
-            raise
+            if not current_app.config['TESTING']:
+                notify_error(user.musicbrainz_row_id, "User has revoked authorization through spotify")
```

Comment:
similar to the above comment - this message goes to a user, so it should be something that they can action - "It seems like you've revoked permission for us to read your spotify account"